docs/rules/enzyme/no-bad-features.md

```diff
@@ -0,0 +1,71 @@
+# Disallows a group of Enzyme features that lead to bad practices. (enzyme/no-bad-features)
+
+This rule restricts usuage of the [Enzyme testing utility](https://airbnb.io/enzyme) by preventing a specific set of features that, more often than not, lead to bad testing practices. 
+
+## Rule Details
+
+The disallowed features are grouped into "immutability" and "component-boundaries" as follows:
+
+#### `immutability`
+- `setState`
+- `setContext`
+- `simulate`
+- `simulateError`
+
+#### `component-boundaries`
+- `instance`
+- `state`
+- `ref`
+
+You can learn more about each of these features on the [Enzyme documentation](https://airbnb.io/enzyme).
+
+Examples of **incorrect** code for the `immutability` group of features:
+
+```ts
+const wrapper = mount(<Foo />);
+wrapper.setState({foo: 'bar'});
+wrapper.setContext({bar: 'baz'});
+wrapper.simulate('click');
+wrapper.simulateError(new Error('Explosions!'));
```

Comment:
I don’t know that these two fit as cleanl into immutability. IMO, they are kind of their own class (maybe separate rule?), and the other two here fit in to component boundaries